pkg/platform/kube/controller/nucliofunction.go

```diff
@@ -151,6 +153,43 @@ func (fo *functionOperator) CreateOrUpdate(ctx context.Context, object runtime.O
 			})
 	}
 
+	//we respond to ready to complete the scale from zero flow. we want to skip flows where once the function
+	// has created or updated and marked as ready, so it will not needlessly run the create or update flow.
+	if functionconfig.FunctionStateInSlice(function.Status.State,
+		[]functionconfig.FunctionState{
+			functionconfig.FunctionStateReady,
+			functionconfig.FunctionStateScaledToZero,
+		}) {
+
+		// make sure this cycle isnt happening after a long run. we want to avoid another "create or update"
+		// that happen as a side effect for updating the function status
+		if function.Status.ScaleToZero != nil &&
+			function.Status.ScaleToZero.LastScaleEventTime != nil &&
+			time.Since(*function.Status.ScaleToZero.LastScaleEventTime) < 60*time.Second {
+
+			if function.Status.State == functionconfig.FunctionStateReady &&
+				(function.Status.ScaleToZero.LastScaleEvent == scalertypes.ResourceUpdatedScaleEvent ||
+					function.Status.ScaleToZero.LastScaleEvent == scalertypes.ScaleFromZeroCompletedScaleEvent) {
+				fo.logger.DebugWithCtx(ctx,
+					"Function was recently deployed, Skipping",
```

Comment:
was recently scaled from zero ?

Reply:
no, that can _also_ happen when a function is recently deployed.
since it is true also for "scaled from zero", the current log line is inherently true